### ISSUE TITLE
pip-compile swallows some error output

### ISSUE THREAD

**Issue by cjerdonek** [jazzband/pip-tools#654]
In some cases, pip-compile seems not to show the full error output from pip.

1. Mac OS X 10.11 
1. Python version: 3.6.5
1. pip version:  9.0.3 
1. pip-tools version: 2.0.1

For example, take the following case where `python setup.py egg_info` fails (trying to install `pyinotify` on Mac OS X 10.11).

The requirements.in file is--

    pyinotify

Running pip-compile gives--

    $ pip-compile --verbose requirements.in 
    Using indexes:
      https://pypi.python.org/simple

                              ROUND 1                           
    Current constraints:
      pyinotify

    Finding the best candidates:
      found candidate pyinotify==0.9.6 (constraint was <any>)

    Finding secondary dependencies:
      pyinotify==0.9.6 not in cache, need to check index
    Traceback (most recent call last):
      File "/.../bin/pip-compile", line 11, in <module>
        sys.exit(cli())
      File "/.../python3.6/site-packages/click/core.py", line 722, in __call__
        return self.main(*args, **kwargs)
      File "/.../python3.6/site-packages/click/core.py", line 697, in main
        rv = self.invoke(ctx)
      File "/.../python3.6/site-packages/click/core.py", line 895, in invoke
        return ctx.invoke(self.callback, **ctx.params)
      File "/.../python3.6/site-packages/click/core.py", line 535, in invoke
        return callback(*args, **kwargs)
      File "/.../python3.6/site-packages/piptools/scripts/compile.py", line 187, in cli
        results = resolver.resolve(max_rounds=max_rounds)
      File "/.../python3.6/site-packages/piptools/resolver.py", line 102, in resolve
        has_changed, best_matches = self._resolve_one_round()
      File "/.../python3.6/site-packages/piptools/resolver.py", line 199, in _resolve_one_round
        for dep in self._iter_dependencies(best_match):
      File "/.../python3.6/site-packages/piptools/resolver.py", line 285, in _iter_dependencies
        dependencies = self.repository.get_dependencies(ireq)
      File "/.../python3.6/site-packages/piptools/repositories/pypi.py", line 146, in get_dependencies
        self._dependencies_cache[ireq] = reqset._prepare_file(self.finder, ireq)
      File "/.../python3.6/site-packages/piptools/_vendored/pip/req/req_set.py", line 634, in _prepare_file
        abstract_dist.prep_for_dist()
      File "/.../python3.6/site-packages/piptools/_vendored/pip/req/req_set.py", line 129, in prep_for_dist
        self.req_to_install.run_egg_info()
      File "/.../python3.6/site-packages/piptools/_vendored/pip/req/req_install.py", line 439, in run_egg_info
        command_desc='python setup.py egg_info')
      File "/.../python3.6/site-packages/piptools/_vendored/pip/utils/__init__.py", line 707, in call_subprocess
        % (command_desc, proc.returncode, cwd))
    pip.exceptions.InstallationError: Command "python setup.py egg_info" failed with error code 1 in /var/folders/q9/j0_5hxt88v5592006s6dd3n80000gs/T/tmpvk6jrc_tbuild/pyinotify/

Notice there is no reason for the InstallationError.

However, if you try to install using pip, you do get info:

    $ pip --version
    pip 9.0.3 from /.../python3.6/site-packages (python 3.6)
    $ pip install pyinotify
    Collecting pyinotify
      Using cached https://files.pythonhosted.org/packages/e3/c0/fd5b18dde17c1249658521f69598f3252f11d9d7a980c5be8619970646e1/pyinotify-0.9.6.tar.gz
        Complete output from command python setup.py egg_info:
        inotify is not available on macosx-10.11-x86_64
    
        ----------------------------------------
    Command "python setup.py egg_info" failed with error code 1 in /private/var/folders/q9/j0_5hxt88v5592006s6dd3n80000gs/T/pip-build-cf9lt18_/pyinotify/

**Comment by cjerdonek** [comment]
Also, this seems to be where `pip-tools`' vendored `pip` would show this extra info: https://github.com/jazzband/pip-tools/blob/3d9e0ecffd2fffbf99ef638300996a61825cdd03/piptools/_vendored/pip/utils/__init__.py#L696

**Comment by cjerdonek** [comment]
I believe this issue will be resolved by this pip PR that I filed: https://github.com/pypa/pip/pull/6339

**Comment by atugushev** [comment]
@cjerdonek awesome!

**Comment by atugushev** [comment]
Now we can see the full error since `pip>=19.1`:
```bash
$ echo pyinotify | pip-compile - -o-
Command errored out with exit status 1:
 command: /Users/albert/Projects/pip-tools/.venv/bin/python3.7 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/var/folders/4x/rz_w89dx1z3bydfd6qpdjn_h0000gn/T/tmpkiq2ri61build/pyinotify/setup.py'"'"'; __file__='"'"'/var/folders/4x/rz_w89dx1z3bydfd6qpdjn_h0000gn/T/tmpkiq2ri61build/pyinotify/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base pip-egg-info
     cwd: /var/folders/4x/rz_w89dx1z3bydfd6qpdjn_h0000gn/T/tmpkiq2ri61build/pyinotify/
Complete output (1 lines):
inotify is not available on macosx-10.14-x86_64
----------------------------------------
Traceback (most recent call last):
  File "/Users/albert/Projects/pip-tools/.venv/bin/pip-compile", line 11, in <module>
    load_entry_point('pip-tools', 'console_scripts', 'pip-compile')()
  File "/Users/albert/Projects/pip-tools/.venv/lib/python3.7/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/Users/albert/Projects/pip-tools/.venv/lib/python3.7/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/Users/albert/Projects/pip-tools/.venv/lib/python3.7/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/albert/Projects/pip-tools/.venv/lib/python3.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/Users/albert/Projects/pip-tools/.venv/lib/python3.7/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/Users/albert/Projects/pip-tools/piptools/scripts/compile.py", line 352, in cli
    results = resolver.resolve(max_rounds=max_rounds)
  File "/Users/albert/Projects/pip-tools/piptools/resolver.py", line 168, in resolve
    has_changed, best_matches = self._resolve_one_round()
  File "/Users/albert/Projects/pip-tools/piptools/resolver.py", line 263, in _resolve_one_round
    their_constraints.extend(self._iter_dependencies(best_match))
  File "/Users/albert/Projects/pip-tools/piptools/resolver.py", line 358, in _iter_dependencies
    dependencies = self.repository.get_dependencies(ireq)
  File "/Users/albert/Projects/pip-tools/piptools/repositories/pypi.py", line 254, in get_dependencies
    download_dir, ireq, wheel_cache
  File "/Users/albert/Projects/pip-tools/piptools/repositories/pypi.py", line 213, in resolve_reqs
    results = resolver._resolve_one(reqset, ireq)
  File "/Users/albert/Projects/pip-tools/.venv/lib/python3.7/site-packages/pip/_internal/legacy_resolve.py", line 359, in _resolve_one
    abstract_dist = self._get_abstract_dist_for(req_to_install)
  File "/Users/albert/Projects/pip-tools/.venv/lib/python3.7/site-packages/pip/_internal/legacy_resolve.py", line 307, in _get_abstract_dist_for
    self.require_hashes
  File "/Users/albert/Projects/pip-tools/.venv/lib/python3.7/site-packages/pip/_internal/operations/prepare.py", line 215, in prepare_linked_requirement
    finder, self.build_isolation,
  File "/Users/albert/Projects/pip-tools/.venv/lib/python3.7/site-packages/pip/_internal/distributions/source.py", line 79, in prepare_distribution_metadata
    self.req.prepare_metadata()
  File "/Users/albert/Projects/pip-tools/.venv/lib/python3.7/site-packages/pip/_internal/req/req_install.py", line 551, in prepare_metadata
    self.run_egg_info()
  File "/Users/albert/Projects/pip-tools/.venv/lib/python3.7/site-packages/pip/_internal/req/req_install.py", line 628, in run_egg_info
    command_desc='python setup.py egg_info')
  File "/Users/albert/Projects/pip-tools/.venv/lib/python3.7/site-packages/pip/_internal/utils/misc.py", line 931, in call_subprocess
    raise InstallationError(exc_msg)
pip._internal.exceptions.InstallationError: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
```

**Comment by atugushev** [comment]
Thanks @cjerdonek!
